src/Components/SearchBar.tsx

```diff
@@ -0,0 +1,220 @@
+import { useEffect, useRef } from 'preact/hooks';
+import { IntlProvider, MarkupText } from 'preact-i18n';
+import t from '../Utility/i18n';
+import Privacy, { PRIVACY_ACTIONS } from '../Utility/Privacy';
+import { searchClient } from '../Utility/search';
+import { rethrow } from '../Utility/errors';
+import { FeatureDisabledWidget } from './FeatureDisabledWidget';
+import type { MergeExclusive } from 'type-fest';
+import type { SearchParams, SearchResponseHit } from 'typesense/lib/Typesense/Documents';
+
+type SearchBarProps = {
+    id: string;
+    placeholder: string;
+    debug?: boolean;
+    style?: string;
+
+    index: string;
+    queryBy?: string;
+    numberOfHits?: number;
+    disableCountryFiltering?: boolean;
+
+    suggestionTemplate?: string | (() => string);
+    emptyTemplate?: string | (() => string);
+    headerTemplate?: string | (() => string);
+    footerTemplate?: string | (() => string);
+
+    filters?: string[];
+} & MergeExclusive<
+    { onAutocompleteSelected: (jqueryEvent: never, hit: Hit, dataset: never) => void },
+    {
+        /** Turn the suggestions into anchors linking to the respective company page. */
+        anchorize: true;
+    }
+>;
+
+type Suggestion = {
+    slug: string;
+    categories?: (keyof typeof window.I18N_DEFINITION.categories)[];
+    name: string;
+    runs?: string[];
+    quality: 'tested' | 'verified' | 'imported' | 'scraped';
+};
+type Hit = SearchResponseHit<Suggestion>;
+
+const countryFilter = (country: typeof window.globals.country) => {
+    const items = ['all', country];
+
+    // Our records often simply specify Germany for companies that are also relevant for Austria and/or Switzerland.
+    // Thus, we explicitly include results from Germany for these countries.
+    //
+    // Ideally, we would rank those additional results lower but as far as I am aware, Typesense doesn't support that.
+    if (['at', 'ch'].includes(country)) items.push('de');
+
+    return `relevant-countries:[${items.join(', ')}]`;
+};
+
+const RealSearchBar = (props: SearchBarProps) => {
+    const input_element = useRef(null);
+    // TODO: We're using an ancient version of autocomplete.js that doesn't have type definitions.
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const autocomplete_ref = useRef<any>(null);
+
+    const {
+        anchorize,
+        debug,
+        disableCountryFiltering,
+        emptyTemplate,
+        filters,
+        footerTemplate,
+        headerTemplate,
+        index,
+        numberOfHits,
+        onAutocompleteSelected,
+        queryBy,
+        suggestionTemplate,
+    } = props;
```

Comment:
You could also just use that as the argument.

Reply:
Really? It doesn't complain to me. To be clear, what I am suggesting looks like this:

```typescript
const RealSearchBar = ({
    id,
    anchorize,
    debug,
    disableCountryFiltering,
    emptyTemplate,
    filters,
    footerTemplate,
    headerTemplate,
    index,
    numberOfHits,
    onAutocompleteSelected,
    queryBy,
    suggestionTemplate,
    placeholder,
    style,
}: SearchBarProps) => {
```

Reply:
Oh okay, I misunderstood. Yeah, that works of course. I don't really have a particular preference either way.